contrib/sarplus/python/pysarplus/SARPlus.py

```diff
@@ -428,3 +434,27 @@ def recommend_k_items_slow(self, test, top_k=10, remove_seen=True):
         )
 
         return self.spark.sql(query)
+
+    def recommend_k_items(
+        self,
+        test,
+        top_k=10,
+        remove_seen=True,
+        use_cache=False,
+        n_user_prediction_partitions=200,
+    ):
+        """Recommend top K items for all users which are in the test set.
+
+        Args:
+            test: test Spark dataframe
+            top_k: top n items to return
+            remove_seen: remove items test users have already seen in the past from the recommended set
+            use_cache: use specified local directory stored in self.cache_path as cache for C++ based fast predictions
+            n_user_prediction_partitions: prediction partitions
```

Comment:
```suggestion
            test (pySpark.DataFrame): test Spark dataframe.
            top_k (int): top n items to return.
            remove_seen (bool): remove items test users have already seen in the past from the recommended set.
            use_cache (bool): use specified local directory stored in `self.cache_path` as cache for C++ based fast predictions.
            n_user_prediction_partitions (int): prediction partitions.
        
        Returns:
            pySpark.DataFrame: Spark dataframe with recommended items.
```

Reply:
Apart from the suggestion (@simonzhaoms please let me know if you can commit FYI @angusrtaylor). 

One question, how much fast is the C++ version vs the Spark version. Maybe it would be good to explain what is happening a little bit. For example, the fast version caches to disk a intermediate computation using C++ and is around X% faster than the standard Spark version

Reply:
@miguelgfierro I'll make another PR to patch your suggestion